### ISSUE TITLE
[JENKINS-25869] Jenkins is getting crashed and shutting down once in a while

### ISSUE THREAD

**Issue by timja-bot** [timja-org/jenkins-gh-issues-poc-11-04#6028]
<p>Our Jenkins master instance got crashed around 3 times in one week and it is getting shutdown by itself once in a while(once or twice per a day). I had mentioned the Environment details in the Environment Field. I am also attaching the last three Jenkins crash logs. This is a serious blocker for our Team, Please let us know what is the issue and remedy.</p>

<p>Please do let me know if you need further details</p>

<p>Quick response will be much useful !! Thanks in advance</p>

<p>Regards,<br/>
Vamsi Hari</p>

---
<details><summary><i>Originally reported by <a title="Vamsi Hari" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=hvamsik">hvamsik</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-25869" target="_blank">Jenkins is getting crashed and shutting down once in a while</a></i></summary>
<i><ul>
<li><b>status</b>: Resolved
<li><b>priority</b>: Blocker
<li><b>resolution</b>: Not A Defect
<li><b>resolved</b>: 2014-12-03T02:09:16+00:00
<li><b>imported</b>: 2025-11-05
</ul></i>
</details>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>This is a bug tracker, not a support site.</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>That said, I'd try to figure out why it crashes in the same proxy related code and investigate the proxy configuration in plugin manager, proxy related environment variables, and/or system properties.</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=hvamsik">hvamsik</a>:</i>
<p>Thanks a lot Daniel for the prompt response,<br/>
Adding to what you said, we are also facing another issue (<a href="https://issues.jenkins-ci.org/browse/JENKINS-25870" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-25870</a>) which is related to proxy settings not working from plugin manager, anything related? <br/>
More Information: Recent config changes that we have done with our Jenkins master is we had added a couple of windows slaves to run selenium Tests on them and we already have a linux slave.</p>

<p>My apologies, if this issue is not related to this Tracker, we just want to corner the issue, It would be so grateful if you can help us in cornering the issue</p>

<p>Please let me know if you need further details,</p>

<p>Below is the part of Tomcat log on the day when earlier crash happened. Below errors are repetitively found in logs (for better understanding i had renamed the hostnames of windows and linux slaves as "windowsslave" & "Linuxslave" )</p>

<p>Nov 19, 2014 2:35:23 PM jenkins.slaves.JnlpSlaveAgentProtocol$Handler$1 onClosed<br/>
WARNING: Channel reader thread: WindowsSlave for + WindowsSlave terminated<br/>
java.io.IOException: Unexpected termination of the channel<br/>
at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:50)<br/>
Caused by: java.io.EOFException<br/>
at java.io.ObjectInputStream$BlockDataInputStream.peekByte(ObjectInputStream.java:2576)<br/>
at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1295)<br/>
at java.io.ObjectInputStream.readObject(ObjectInputStream.java:349)<br/>
at hudson.remoting.Command.readFrom(Command.java:92)<br/>
at hudson.remoting.ClassicCommandTransport.read(ClassicCommandTransport.java:71)<br/>
at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:48)<br/>
Nov 19, 2014 2:35:33 PM hudson.TcpSlaveAgentListener$ConnectionHandler run<br/>
INFO: Accepted connection #12 from /51.16.108.187:61593<br/>
Nov 19, 2014 2:35:34 PM hudson.TcpSlaveAgentListener$ConnectionHandler run<br/>
WARNING: Connection #12 failed<br/>
java.io.EOFException: unexpected stream termination<br/>
at hudson.remoting.ClassicCommandTransport.create(ClassicCommandTransport.java:135)<br/>
at hudson.remoting.Channel.<init>(Channel.java:406)<br/>
at hudson.remoting.Channel.<init>(Channel.java:402)<br/>
at hudson.remoting.Channel.<init>(Channel.java:363)<br/>
at hudson.remoting.Channel.<init>(Channel.java:359)<br/>
at hudson.slaves.SlaveComputer.setChannel(SlaveComputer.java:344)<br/>
at jenkins.slaves.JnlpSlaveAgentProtocol$Handler.jnlpConnect(JnlpSlaveAgentProtocol.java:122)<br/>
at jenkins.slaves.JnlpSlaveAgentProtocol2$Handler2.run(JnlpSlaveAgentProtocol2.java:99)<br/>
at jenkins.slaves.JnlpSlaveAgentProtocol2.handle(JnlpSlaveAgentProtocol2.java:43)<br/>
at hudson.TcpSlaveAgentListener$ConnectionHandler.run(TcpSlaveAgentListener.java:154)</p>

<p>Nov 19, 2014 2:39:45 PM hudson.model.Run execute<br/>
INFO: servicegateway-static-build-git-claimsource #520 main build action completed: SUCCESS</p>

<p>Nov 19, 2014 2:39:54 PM hudson.remoting.Channel$2 handle<br/>
SEVERE: Failed to execute command Pipe.EOF(272) (channel Linuxslave)<br/>
java.lang.NullPointerException<br/>
at hudson.remoting.ProxyWriter$EOF.execute(ProxyWriter.java:177)<br/>
at hudson.remoting.Channel$2.handle(Channel.java:461)<br/>
at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:60)<br/>
Nov 19, 2014 2:39:54 PM hudson.remoting.Channel$2 handle<br/>
SEVERE: This command is created here<br/>
Command Pipe.EOF(272) created at<br/>
at hudson.remoting.Command.<init>(Command.java:67)<br/>
at hudson.remoting.Command.<init>(Command.java:50)<br/>
at hudson.remoting.ProxyWriter$EOF.<init>(ProxyWriter.java:169)<br/>
at hudson.remoting.ProxyWriter.close(ProxyWriter.java:124)<br/>
at hudson.remoting.ProxyWriter.finalize(ProxyWriter.java:132)<br/>
at java.lang.ref.Finalizer.invokeFinalizeMethod(Native Method)<br/>
at java.lang.ref.Finalizer.runFinalizer(Finalizer.java:83)<br/>
at java.lang.ref.Finalizer.access$100(Finalizer.java:14)<br/>
at java.lang.ref.Finalizer$FinalizerThread.run(Finalizer.java:172)<br/>
Nov 19, 2014 2:40:22 PM hudson.remoting.Channel$2 handle<br/>
SEVERE: Failed to execute command Pipe.EOF(275) (channel Linuxslave)<br/>
java.lang.NullPointerException<br/>
at hudson.remoting.ProxyWriter$EOF.execute(ProxyWriter.java:177)<br/>
at hudson.remoting.Channel$2.handle(Channel.java:461)<br/>
at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:60)</p>


<p>Regards,<br/>
Vamsi Hari</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>That looks like a completely different issue. What is the relation of that to the earlier reported JRE crashes?</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=hvamsik">hvamsik</a>:</i>
<p>I am not sure, I had added that as you said you saw some proxy related code in the error log. But from every JVM crash log there are some logs related svn and proxy. BTW We are still having our Jenkins instance crashed regularly, we had recent crash on Jan 1. I am attaching the recent crash log too.</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=hvamsik">hvamsik</a>:</i>
<p>crash log on Jan 1 2015</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>Please provide the configuration of the job <tt>PortalEXT-EAR-deployment-PEO</tt> and <tt>PortalEXT-myts-static-STATIC-deployment</tt> and the output on the <tt>/systemInfo</tt> URL. Or better yet, install <a href="https://wiki.jenkins-ci.org/display/JENKINS/Support+Core+Plugin" class="external-link" target="_blank" rel="nofollow noopener">https://wiki.jenkins-ci.org/display/JENKINS/Support+Core+Plugin</a> and attach a support bundle to this issue (in addition to the two job configurations, since they're not in the bundle).</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=hvamsik">hvamsik</a>:</i>
<p>Daniel - I had attached all the info requested by you. FYI, In the job config files i had commented host names and mentioned respective details over there.  </p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>I'd try to make sure to unset the <tt>http_proxy</tt> and related environment variables before starting the Jenkins process (or rather Tomcat in your case?), instead (if necessary) using proxy configuration in Jenkins (Manage Jenkins » Manage Plugins » Advanced). This may result in child processes (e.g. shell scripts and the tools they run) not inheriting the proxy configuration by default though.</p>

<p>The bug may well be in the gconf (Gnome) library or the JDK, as that's where the crash occurs.<br/>
<a href="http://hg.openjdk.java.net/jdk7u/jdk7u/jdk/file/54acd5cd0485/src/solaris/native/sun/net/spi/DefaultProxySelector.c#l155" class="external-link" target="_blank" rel="nofollow noopener">http://hg.openjdk.java.net/jdk7u/jdk7u/jdk/file/54acd5cd0485/src/solaris/native/sun/net/spi/DefaultProxySelector.c#l155</a></p>

<p>Please also note that the user and password to the proxy are included in plain text in the support bundle (nodes/master/environment.txt), so you should change those now.</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=hvamsik">hvamsik</a>:</i>
<p>Thanks Daniel for your help,<br/>
  I will try removing all the proxy related environment variables and restarting the Jenkins container(Tomcat) and let you know about the result.</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>This appears to be a bug in your JDK, and has nothing to do with Jenkins (except that it runs code that causes the JDK code to run).</p>

<p>See:<br/>
<a href="http://bugs.java.com/bugdatabase/view_bug.do?bug_id=7188755" class="external-link" target="_blank" rel="nofollow noopener">http://bugs.java.com/bugdatabase/view_bug.do?bug_id=7188755</a><br/>
<a href="https://bugs.launchpad.net/ubuntu/+source/openjdk-6/+bug/885195" class="external-link" target="_blank" rel="nofollow noopener">https://bugs.launchpad.net/ubuntu/+source/openjdk-6/+bug/885195</a><br/>
<a href="http://mail.openjdk.java.net/pipermail/jdk7u-dev/2012-August/003911.html" class="external-link" target="_blank" rel="nofollow noopener">http://mail.openjdk.java.net/pipermail/jdk7u-dev/2012-August/003911.html</a></p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=hvamsik">hvamsik</a>:</i>
<p>Thanks Daniel for your help,<br/>
We will keep posted if we find any solution for this java bug.</p>

<p>Cheers,<br/>
Vamsi Hari</p>
